src/laddie/laddie_main.f90

```diff
@@ -23,6 +24,47 @@ MODULE laddie_main
     
 ! ===== Main routines =====
 ! =========================
+
+  SUBROUTINE run_laddie_model( mesh, ice, ocean, BMB, time)
+    ! Run the laddie model
+
+    ! In- and output variables
+
+    TYPE(type_mesh),                        INTENT(IN)    :: mesh
+    TYPE(type_ice_model),                   INTENT(IN)    :: ice
+    TYPE(type_ocean_model),                 INTENT(IN)    :: ocean
+    TYPE(type_BMB_model),                   INTENT(INOUT) :: BMB
+    REAL(dp),                               INTENT(IN)    :: time
+
+    ! Local variables:
+    CHARACTER(LEN=256), PARAMETER                         :: routine_name = 'run_laddie_model'
+    INTEGER                                               :: vi
+    REAL(dp), PARAMETER                                   :: T_off  = 0.0_dp  ! [degC] Initial temperature offset
+    REAL(dp), PARAMETER                                   :: S_off  = -0.1_dp ! [PSU]  Initial salinity offset
+ 
+    ! Add routine to path
+    CALL init_routine( routine_name)
+ 
+    ! Initialise ambient T and S
+    CALL compute_ambient_TS( mesh, BMB%laddie, ocean, ice)
```

Comment:
A new feature I've been working on is the option to use assertions: optionally-compiled checks on input data. E.g., I presume Laddie only works if the ambient ocean temperature is above freezing? Then you could add:

#if (DO_ASSERTIONS)
    call assert( test_ge( ocean%T, ocean%T_freeze), 'ocean temperatures below freezing detected')
#endif

Here, 'DO_ASSERTIONS' is a compiler flag which right now is by default set to true, but can be optionally set to false (which would remove these safety checks and thereby speed up the model). 'assert( condition, fail_message)' causes the model to crash if condition == .false., displaying the fail_message. 'test_ge( a,b)' is basically just a wrapper for 'a>b'.

Safety checks like this already exist in a lot of places in the model, but they are often quite slow, so wrapping them inside DO_ASSERTIONS allows you to compile the model without them.

Reply:
That's great. Right now it's not trivial for me to list assertions while 'dumping' the code. But I'll need to add these throughout the ocean and laddie code.

I'll open an issue to add these with a clear head.